speaking-portal-project/src/animationFactory/animationProcessor.ts

```diff
@@ -0,0 +1,155 @@
+import { MouthCue, MouthCueArray } from '../types'
+import fs from 'fs'
+
+export async function mouthCuesToInputFile({
+    characterSelect,
+    mouthCues,
+    outputPath = './tmp/input.txt',
+}: {
+    characterSelect: string
+    mouthCues: MouthCue[]
+    outputPath?: string
+}) {
+    try {
+        await fs.promises.writeFile(outputPath, generateFrameData(characterSelect, mouthCues), {
+            flag: 'w',
+        })
+        /* Lines 17-20 for testing purposes only
+        const contents = await fs.promises.readFile(outputPath, 'utf-8')
+        console.log(contents);
+
+        return contents
+        */
+    } catch (err) {
+        console.log(err)
+        return Error('MouthCuesFileError')
+    }
+}
+
+/*
+This converts  data from the MouthCue type into a format for ffmpeg
+*/
+export function generateFrameData(avatar: string, mouthCues: MouthCue[]) {
+    let frameData = ''
+    let blink = false
+    let lastBlink = -1
+    let lastPoseChange = -1
+    let idle = true
+    let breathPhase = 0 // 0 = rest, 1 = inhaling, 2 = lungs full, 3 = exhaling
+
+    console.log("Adding realism...")
+    for (const mouthCuesKey in mouthCues) {
+        // Duration of frame is calculated based on the mouth cue
+        let frameDur = (mouthCues[mouthCuesKey].end - mouthCues[mouthCuesKey].start).toFixed(2)
+        let currentSec = parseInt(mouthCues[mouthCuesKey].start.toFixed(0))
+
+        // TODO: use for testing. After confirming that blinking and other realism is good, delete this
+        console.log("t=%d, frameDur: %d, lastBlink: %d, idling? %s, lastPoseChange: %d",
+                    currentSec, frameDur, lastBlink, lastPoseChange, idle, lastPoseChange)
+
+        // Start building frame's path
```

Comment:
We should eventually extract this into a separate function, but not needed for now!

Reply:
Yeah, I think making it a separate function will need to wait until we have all the realism pieces together.. unless you have an idea for how to implement it now!